File: src/modlunky2/sprites/full_sheets/decorations.py
from pathlib import Path

from modlunky2.sprites.base_classes.base_json_sprite_merger import BaseSpriteMerger
from modlunky2.sprites.deco_sheet import CaveDecoSheet


class UdjatWallHeads(BaseSpriteMerger):
    _target_sprite_sheet_path = Path(
        "Data/Textures/Entities/Decorations/udjat_wall_heads.png"
    )
    _grid_hint_size = 8
    _origin_map = {
        CaveDecoSheet: {"udjat_wall_heads": (0, 0, 4, 4)},
    }

File: crates/ml2_sprites/src/data/mergers/decorations.rs
// Sprite loader/merger configs describing which chunks live in which
// sheet, keyed by tile name for the sprite fetcher lookup.

use crate::data::chunks;
use crate::{MergerConfig, OriginEntry};

pub fn udjat_wall_heads() -> MergerConfig {
    MergerConfig {
        target_sprite_sheet_path: "Data/Textures/Entities/Decorations/udjat_wall_heads.png".into(),
        grid_hint_size: 8,
        origin_map: vec![OriginEntry {
            loader_name: "CaveDecoSheet".into(),
            target_chunks: chunks(&[("udjat_wall_heads", 0.0, 0.0, 4.0, 4.0)]),
        }],
    }
}
